### DISCUSSION TITLE
temperature label rename

### DISCUSSION THREAD

**Discussion by sp2003** [arendst/Tasmota#15060] [Q&A] [answered]
hi

it's possible to rename label temperature?
![IMG_20220307_221016](https://user-images.githubusercontent.com/34820831/157118945-813dbaf1-d48f-4bf4-a5b2-62709abceaa6.jpg)

**Comment by barbudor** [comment, marked as answer]
No
Why would you call temperature something else than temperature ?

Remeber that from Tasmota perspective, the Web UI is mainly a configuration and debug tool and not intended as an everyday-use.
Dashboard are generally hosted on a home-automation software from the data published over MQTT.

**Reply by sp2003** [reply]
I know that the ui web is for configuration and debug, but I was hoping it was possible as for button labels.
However, sometimes it is faster to access to tasmota web UI of and command a relay instead of opening your domotic hub, so renaming the sensor labels might be helpful.

**Reply by barbudor** [reply]
You may investigate using [Scripting-Language](https://tasmota.github.io/docs/Scripting-Language/) as it allows to add some elements to the Web UI
However these will be in addition to the existing lines. I don't think you can totally remove the default Sensor display.

**Reply by sfromis** [reply]
You can get different names by setting the build time option `DS18x20_USE_ID_AS_NAME` but that just avoids the numbers to instead use the 3 last bytes of the sensor id. In general, Tasmota has no features for customizing sensor names, as this would mean having lots and lots of potentially assignable strings, which there is no room for in the very limited settings area.

**Reply by ffrediani** [reply]
Well, anyone may have any context that requires it to be changed to call it whatever else. What is the harm on that ?

Even though the Web IO is mainly for configuration, for simpler applications or prototypes it is fine to visualize readings there in a nicer way with proper labels to avoid confusion.

Having to recompile seems overkill and would be nice to have simple console command that allows to rename each pin display in the main dashboard.

**Reply by sfromis** [reply]
The "harm" of such a feature would be that the available space is too limited for having dynamic configuration strings for all possible sensor names.

However, for DS18B20 specifically, a feature was implemented, which you would have found if you actually had read the documentation.
https://tasmota.github.io/docs/DS18x20/#multiple-sensors

**Reply by ffrediani** [reply]
Hi
Even having some "Alias" command to specific the sensor and input would that use that much space either in the flash or memory that is significant ? For ESP32 seems there is a fair amount of space, not entirely sure for ESP8266.

**Comment by HofeBY2** [comment]
search for
Address in logfile
Example command
DS18Alias B33C6CE347110815 yourSensorName

**Reply by ffrediani** [reply]
Thanks. It seems Tasmota has to be compiled with this option and the vanilla version isn't.

**Reply by HofeBY2** [reply]
yes, option "#define DS18x20_USE_ID_ALIAS" has to be set for compiling.
// Add support aliasing for DS18x20 sensors. See comments in xsns_05 files (+0k5 code)

It´s a good idea to compile your own tasmota version.
